test/screens/screens_by_alert_test.exs

```diff
@@ -1,35 +1,73 @@
 defmodule Screens.ScreensByAlertTest do
   @moduledoc false
-  use ExUnit.Case, async: true
+  use ExUnit.Case
   alias Screens.ScreensByAlert
 
-  describe "start_link/1" do
-    test "returns {:ok, pid} when start_link is called" do
-      Application.stop(:screens)
+  setup do
+    Application.stop(:screens)
 
-      on_exit(fn ->
-        Application.start(:screens)
-      end)
+    on_exit(fn ->
+      Application.start(:screens)
+    end)
+  end
 
-      assert {:ok, _pid} = ScreensByAlert.start_link()
+  describe "start_link/1" do
+    test "returns {:ok, pid} when start_link is called" do
+      assert {:ok, pid} = ScreensByAlert.start_link([])
+      on_exit(fn -> Process.exit(pid, :done) end)
     end
   end
 
   describe "put_data/2" do
     test "returns :ok" do
-      assert :ok = ScreensByAlert.put_data(0, [])
+      {:ok, pid} = ScreensByAlert.start_link([])
+      assert :ok = ScreensByAlert.put_data(1, [1])
+      on_exit(fn -> Process.exit(pid, :done) end)
     end
   end
 
   describe "get_screens_by_alert/1" do
```

Comment:
Could you add one more test checking that individual screens expire from an alert key over time even if the key itself doesn't expire?

Reply:
Yeah good thinking. I really worry about these `sleep` calls causing flakiness, but I'm not sure of a way around it.